### PR TITLE
[chassis] test_memory_checker.py use same random dut for fixtures and functions 

### PR DESCRIPTION
**Pull request by wenyiz2021**
### Description of PR

This is testcase enhancement, it'll save number of `skipped` testcases in this test file.
call combination of `enum_rand_one_per_hwsku_frontend_hostname, enum_rand_one_asic_index, enum_dut_feature` to let random feature choosing same dut as enum_rand_one_per_hwsku_frontend_hostname. 
original way of `enum_dut_feature_container, enum_rand_one_per_hwsku_frontend_hostname` will generate combination of 
`test_memory_checker[lc7-1-lc5-1|lldp1` which `lc5-1|lldp1` is generated by `enum_dut_feature_container` and `enum_rand_one_per_hwsku_frontend_hostname` is `lc7`. This combination then gets skipped when decoding enum_dut_feature_container. This PR intends to avoid those combinations, instead of skip.

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [x] 202205

#### How did you verify/test it?
Before lots of skips due to:
```
SKIPPED [4] /azp/_work/35/s/tests/common/helpers/assertions.py:13: Skips testing memory_checker of container 'teamd' on the DuT 'lc7-1' since another DuT 'lc5-1' was chosen.
============== 1 failed, 7 passed, 312 skipped in 4599.02 seconds ==============
```

After, only skipped when container is not telemetry:
```
=========================== short test summary info ============================
SKIPPED [1] /var/src/private/sonic-mgmt-int/tests/common/helpers/assertions.py:13: Skips testing memory_checker of container 'teamd0' since memory monitoring is only enabled for 'telemetry'.
...
==================== 2 passed, 26 skipped in 986.11 seconds ====================
```

### CODE REVIEW

File: tests/memory_checker/test_memory_checker.py [hunk 1]
```diff
@@ -224,9 +224,14 @@ def remove_stress_utility(duthost, container_name):
     logger.info("BGP sessions are started up.")
 
 
+@pytest.fixture(autouse=True)
```
Comment (arlakshm) [inline tests/memory_checker/test_memory_checker.py:227]: how does this help? `enum_rand_one_per_hwsku_frontend_hostname` is module fixture buit rand_dut_for_all_functions is function level fixture, how does this work?

File: tests/memory_checker/test_memory_checker.py [hunk 2]
```diff
@@ -224,9 +224,14 @@ def remove_stress_utility(duthost, container_name):
...
+def rand_dut_for_all_functions(enum_rand_one_per_hwsku_frontend_hostname):
+    return enum_rand_one_per_hwsku_frontend_hostname
+
+
 @pytest.fixture
 def test_setup_and_cleanup(duthosts, creds, enum_dut_feature_container,
```
Comment (arlakshm) [inline tests/memory_checker/test_memory_checker.py:233]: iso modifying the test, can we update `generate_dut_feature_container_list` to generate the containers only the choosen dut.

**Review by arlakshm — COMMENTED** [review]
as comments

File: tests/memory_checker/test_memory_checker.py [hunk 3]
```diff
@@ -268,8 +264,8 @@ def test_setup_and_cleanup(duthosts, creds, enum_dut_feature_container,
 
 
 @pytest.fixture
-def remove_and_restart_container(duthosts, creds, enum_dut_feature_container,
-                                 enum_rand_one_per_hwsku_frontend_hostname):
+def remove_and_restart_container(duthosts, creds, enum_rand_one_per_hwsku_frontend_hostname,
+                                 enum_rand_one_asic_index, enum_dut_feature):
     """Removes and restarts 'telemetry' container from DuT.
 
     Args:
```
Comment (SuvarnaMeenakshi) [inline tests/memory_checker/test_memory_checker.py:271]: Minor comment: could change Args in the comment to reflect new list

Comment (wenyiz2021) [reply tests/memory_checker/test_memory_checker.py:275]: done, thanks!

**Review by SuvarnaMeenakshi — APPROVED** [review]
lgtm

**Comment by aaronpayment** [comment]
Hi @wenyiz2021 

This change conflicts with the in progress review that you are helping review at https://github.com/sonic-net/sonic-mgmt/pull/9193

In the refactor review https://github.com/sonic-net/sonic-mgmt/pull/9193 we are already trying to reduce the skip combinations and in a recent T2 test run we had the test cases are

```

2023-08-05 22:04:30 | passed | memory_checker/test_memory_checker.py::test_memory_checker::test_memory_checker[telemetry-cmp210-3]
2023-08-05 22:12:53 | passed | memory_checker/test_memory_checker.py::test_memory_checker::test_memory_checker_recover[telemetry-cmp210-3]
2023-08-05 22:18:32 | passed | memory_checker/test_memory_checker.py::test_memory_checker::test_monit_reset_counter_failure[telemetry-cmp210-3]
2023-08-05 22:32:51 | passed | memory_checker/test_memory_checker.py::test_memory_checker::test_memory_checker_without_container_created[telemetry-cmp210-3]
2023-08-05 22:38:45 | passed | memory_checker/test_memory_checker.py::test_memory_checker::test_memory_checker[telemetry-cmp210-5]
2023-08-05 22:44:47 | passed | memory_checker/test_memory_checker.py::test_memory_checker::test_memory_checker_recover[telemetry-cmp210-5]
2023-08-05 22:50:27 | passed | memory_checker/test_memory_checker.py::test_memory_checker::test_monit_reset_counter_failure[telemetry-cmp210-5]
2023-08-05 23:03:45 | passed | memory_checker/test_memory_checker.py::test_memory_checker::test_memory_checker_without_container_created[telemetry-cmp210-5]
```

Are these sufficient test cases and we can abandon the changes in your review here?

**Pull request merged**